### PR TITLE
Fix NullPointerException when clicking an event

### PR DESCRIPTION
**Pull request by thellmund**
When clicking an event in the week view, the corresponding `CalendarItem` is retrieved from the database. If the query returns null for whatever reason, the app crashes (happened very rarely). I’ve added a null check to prevent this.

Additionally, instead of using the event's start and end time to query for the `CalendarItem`, the proper event ID is now used.

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarDao.java [hunk 1]
```diff
@@ -70,10 +70,6 @@
             "ORDER BY dtend, dtstart LIMIT 4")
     List<CalendarItem> getNextCalendarItems();
 
-    // TODO: migrate dtstart and dtend to be proper dates
-    @Query("SELECT * FROM calendar WHERE dtstart LIKE '%' || :start AND dtend LIKE '%' || :end ")
-    CalendarItem getCalendarItemByStartAndEndTime(String start, String end);
-
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarDao.java]: good riddance

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarController.java [hunk 2]
```diff
@@ -186,10 +186,9 @@ public void deleteLectureFromBlacklist(int widgetId, String lecture) {
         return lectures;
     }
 
-    public CalendarItem getCalendarItemByStartAndEndTime(DateTime start, DateTime end) {
-        String startString = DateTimeUtils.INSTANCE.getDateTimeString(start);
-        String endString = DateTimeUtils.INSTANCE.getDateTimeString(end);
-        return calendarDao.getCalendarItemByStartAndEndTime(startString, endString);
+    @Nullable
+    public CalendarItem getCalendarItemById(String id) {
+        return calendarDao.getCalendarItemById(id);
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarController.java]: IMHO `getCalendarItemById()` shouldn't be nullable, but I guess you can't guarantee that that ID exists.
Also I think the id should probably be an int :man_shrugging:

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java [hunk 3]
```diff
@@ -438,9 +438,16 @@ public void onEditEvent(@NotNull CalendarItem calendarItem) {
...
-                new DateTime(weekViewEvent.getStartTime()),
-                new DateTime(weekViewEvent.getEndTime()));
+        String eventId = weekViewEvent.getIdentifier();
+        if (eventId == null) {
+            return;
+        }
+
+        CalendarItem item = calendarController.getCalendarItemById(eventId);
+        if (item == null) {
+            return;
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java]: This shoudn't happen, if everything was set up correctly.
Could we add some logging here, so we maybe get a chance to debug this eventually?

File: app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java [hunk 4]
```diff
@@ -269,7 +269,7 @@ private void refreshWeekView() {
             // Lets change some dimensions to best fit the view.
             mWeekView.setTextSize((int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, 16, getResources().getDisplayMetrics()));
             mWeekView.setEventTextSize((int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_SP, 12, getResources().getDisplayMetrics()));
-            mWeekView.setXScrollingSpeed(0.4f);
+            mWeekView.setXScrollingSpeed(0.5f);
```
Comment (kordianbruck) [inline app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java]: any more info on this change?

Comment (thellmund) [reply app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarActivity.java]: Oops, somehow slipped in there. I was just testing out something for myself and forgot to remove it.

File: app/src/main/res/layout/activity_calendar.xml [hunk 5]
```diff
@@ -43,7 +43,7 @@
                     app:noOfVisibleDays="1"
                     app:textSize="14sp"
                     app:todayBackgroundColor="#1848adff"
-                    app:xScrollingSpeed="0.4" />
+                    app:xScrollingSpeed="0.5" />
```
Comment (kordianbruck) [inline app/src/main/res/layout/activity_calendar.xml]: Also here: this slipped in

**Pull request merged**